### PR TITLE
adding k8s artifacts

### PR DESCRIPTION
**Pull request by dchinn1**
Created basic artifacts to deploy jwt-inspect. Includes optional, loadable custom css.

Make targets created for easy use.

### CODE REVIEW

File: deployments/base/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations:
+  name: jwtinspect
```
Comment (xytian315) [inline deployments/base/deployment.yaml]: nitpick: our repo is called jwt-inspect, maybe here is called jwt-inspect too same as other sections referring  `jwtinspect` in this file

File: deployments/base/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+images:
+- name: ghcr.io/openmicrotools/jwt-inspect
+  newTag: latest
+
+metadata:
+  annotations:
+    name: jwt-inspect
+  labels:
+    app: jwt-inspect
+namespace: open-microtools
```
Comment (xytian315) [inline deployments/base/kustomization.yaml]: nitpick: probably still better just call jwt-inspect

File: deployments/base/namespace.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: open-microtools
```
Comment (xytian315) [inline deployments/base/namespace.yaml]: missing end of line here

File: deployments/base/service.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+  labels:
+    app: jwtinspect
```
Comment (xytian315) [inline deployments/base/service.yaml]: same as above align with one naming convention crossing the whole is better.  `jwt-inspect`

File: deployments/overlays/ingress/ingress.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+kind: Ingress
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: jwtinspect
```
Comment (xytian315) [inline deployments/overlays/ingress/ingress.yaml]: same comment as naming convention in this file

File: deployments/overlays/nodeport/kustomization.yaml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+resources:
```
Comment (xytian315) [inline deployments/overlays/nodeport/kustomization.yaml:3]: question: where is nodeport customize stuff? 
README also empty. is that intentional?

File: deployments/overlays/ingress/kustomization.yaml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+# Uncomment to add custom css as a configmap
+# - ../../css 
+- ingress.yaml
+
+metadata:
+  annotations:
+    name: jwt-inspect
+  labels:
+    app: jwt-inspect
+namespace: open-microtools
```
Comment (xytian315) [inline deployments/overlays/ingress/kustomization.yaml]: I think these section is probably not necessary to have it here. the folder is meant for overlay ingress

File: deployments/base/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+  labels:
+    app: jwtinspect
+    version: v1
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: jwtinspect
+      version: v1
```
Comment (xytian315) [inline deployments/base/deployment.yaml:13]: question: do we need version: v1 label here. what is the intention?

File: Makefile [hunk 9]
```diff
@@ -58,3 +58,13 @@ run: wasm
 .PHONY: docker-build
 docker-build:
 	docker build -f ./docker/Dockerfile . --tag jwt-server:$(IMAGE_TAG)
+
+# Kustomize targets
```
Comment (xytian315) [inline Makefile:62]: question: does it make sense to just create a normal deploy (default serveType: clusterIP)?
```
deploy: 
   kustomize build deployments | kubectl apply -f -
```
this is probably the most comment scenario when people just develop something and experience with it.

File: deployments/overlays/nodeport/kustomization.yaml (hunk 6 above)
Comment (xytian315) [reply deployments/overlays/nodeport/kustomization.yaml:3]: we can delete nodeport folder if we don't want this.

**Comment by xytian315** [comment]
great work !in the future, we can create a script to set up kind env too. easier to track the kind/ k8s version etc.

File: config.yaml [hunk 10]
```diff
@@ -0,0 +1,10 @@
+apiVersion: kind.x-k8s.io/v1alpha4
```
Comment (xytian315) [inline config.yaml]: hmm I am not sure this is a good file name and a good place to put this file.

File: deployments/overlays/base/README.md [hunk 11]
```diff
@@ -0,0 +1,4 @@
+# Nodeport Kustomization
```
Comment (xytian315) [inline deployments/overlays/base/README.md]: question: why do we need the folder `deployments/overlays/base`?  we already have `deployments/base`

If you wanna have some common labels and common annotations. I think you can use `commonLabels` and `commonAnnotations:` in `deployment/base/kustomization.yaml` 

I don't feel like we quite need this folder.

File: deployments/overlays/nodeport/kustomization.yaml [hunk 12]
```diff
@@ -1,7 +1,20 @@
...
+images:
+- name: ghcr.io/openmicrotools/jwt-inspect
+  newTag: latest
+
+metadata:
+  annotations:
+    name: jwt-inspect
+  labels:
+    app: jwt-inspect
+namespace: open-microtools
```
Comment (xytian315) [inline deployments/overlays/nodeport/kustomization.yaml:17]: I feel like we don't need from line #12 to #17. If we want to have commonLabels and commonAnnotations we can have it defined in deployment/base folder.

File: deployments/overlays/nodeport/manifest.yaml [hunk 13]
```diff
@@ -0,0 +1,60 @@
+apiVersion: v1
```
Comment (xytian315) [inline deployments/overlays/nodeport/manifest.yaml]: hmm is this file generated in some of the steps you were testing?

File: deployments/overlays/nodeport/kustomization.yaml (hunk 12 above)
Comment (dchinn1) [reply deployments/overlays/nodeport/kustomization.yaml:17]: Having at least namespace here allows users to make a single change to change all the namespaces, same for the annotations/labels. We can do it either way, but I thought having it here made it easier for the user.

File: deployments/overlays/base/README.md (hunk 11 above)
Comment (dchinn1) [reply deployments/overlays/base/README.md]: I have this folder here to apply all the same kustomizations we have in the other two overlays, and make the usage more consistent. I'm actually don't have namespace listed in either of the configurations, so applying from base will put it in default.

File: deployments/base/deployment.yaml (hunk 8 above)
Comment (dchinn1) [reply deployments/base/deployment.yaml:13]: We don't, so I can just remove it

File: deployments/overlays/nodeport/manifest.yaml (hunk 13 above)
Comment (dchinn1) [reply deployments/overlays/nodeport/manifest.yaml]: Removed file

File: deployments/overlays/base/README.md (hunk 11 above)
Comment (xytian315) [reply deployments/overlays/base/README.md]: >I have this folder here to apply all the same kustomizations we have in the other two overlays, and make the usage more consistent.

I think this is exactly the use case for deployment/base folder. 

I don't see a reason why you can't define namespace in its own respective folder.  User either choose ingress or nodeport, you can't apply both. I don't think there is any user needed to have a place to change and apply both ingress and nodeport. Especially in this folder/file here, only common label/annotation/namespace stuff, I don't think there is any use cases if you define a service in namespace defined by base, and user want to apply ingress in a different namespace.

File: deployments/overlays/nodeport/kustomization.yaml (hunk 12 above)
Comment (xytian315) [reply deployments/overlays/nodeport/kustomization.yaml:17]: similar as the other thread, I don't know if user will ever want to apply Ingress in a different namespace than base defined where service is. and I think ingress should have the same common labels and annotations as base folder.

**Review by xytian315 — APPROVED** [review]

**Pull request merged**